src/Build/BackEnd/Components/RequestBuilder/AssemblyLoadsTracker.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+//
+
+using System;
+using System.Collections.Concurrent;
+using Microsoft.Build.BackEnd.Logging;
+using Microsoft.Build.Framework;
+
+namespace Microsoft.Build.BackEnd.Components.RequestBuilder
+{
+    internal class AssemblyLoadsTracker : MarshalByRefObject, IDisposable
+    {
```

Comment:
Does this type run on both core and framework? Heavy use of `AppDomain` makse me think framework but not seeing `#if` to remove it from core builds.

Reply:
It's core and framework tested.
That being said - the dictionary with app domains is really needed just on FW - so putting this behing `#ifdef`
Rest I believe needs to stay as is - but it might be my lack of knowledge - for tracking assembly loads I need `AppDomain` (even if it's just the default one on core) to subscribe - correct?